houston/mission.py

```diff
@@ -55,14 +61,27 @@ def extended(self, cmd: Command) -> 'Mission':
         end.
         """
         cmds = self.commands + (cmd,)
-        return Mission(self.environment, self.initial_state, cmds)
+        return Mission(self.environment, self.initial_state, cmds, self.system)
 
     def to_dict(self) -> Dict[str, Any]:
         return {
             'configuration': self.configuration.to_dict(),
             'environment': self.environment.to_json(),
             'initial_state': self.initial_state.to_json(),
-            'commands': [c.to_json() for c in self.commands]}
+            'commands': [c.to_json() for c in self.commands],
+            'system': self.system.name}
+
+    def run(self, bz: BugZooClient, snapshot: str) -> 'MissionOutcome':
```

Comment:
Minor nitpick: can we rename `snapshot` to `snapshot_name`, or can we rename to `snapshot_or_name` and change the type to `Union[str, Snapshot`]?